src/main/java/com/example/simpleannotation/model/BuilderGeneratedClass.java

```diff
@@ -0,0 +1,181 @@
+                annotatedClass.getClassToBuildPackageName(),
+                annotatedClass.getClassToBuild()));
+        this.getConstructors().forEach(constructor ->
+            constructor.addCodeLine("this.container = new HashMap<>();"));
+    }
+
+    private void addSingletonPattern() {
+        this.getAttributes().add(new AttributeDescriptor(INSTANCE,
+                computeBuilderName(),
+                "private static"));
+        MethodDescriptor descriptor = new MethodDescriptor("getInstance", computeBuilderName());
+        descriptor.addModifier(PUBLIC);
+        descriptor.addModifier("static");
+        descriptor.addCodeLine("if (instance == null) {")
+                .addCodeLine(String.format("\tinstance = new %s();", computeBuilderName()))
+                .addCodeLine("}")
+                .addCodeLine("return instance;");
+        this.getMethods().add(descriptor);
+        ConstructorDescriptor privateConstructor = new ConstructorDescriptor(computeBuilderName());
+        privateConstructor.addModifier(PRIVATE);
+        this.getConstructors().add(privateConstructor);
+    }
+
+    private void addSimpleConstructor() {
+        ConstructorDescriptor publicConstructor = new ConstructorDescriptor(computeBuilderName());
+        publicConstructor.addModifier(PUBLIC);
+        this.getConstructors().add(publicConstructor);
+    }
+
+    private void addFluentSetters () {
+        for (Map.Entry<AttributeDescriptor, Optional<MethodDescriptor>> attributeMethodMapping :
+             this.annotatedClass.getAttributeSetterMapping().entrySet()) {
+            this.getMethods().add(new MethodDescriptor(
+                    computeSetterName(attributeMethodMapping.getKey().getName()),
+                    computeBuilderName())
+                    .addModifier(PUBLIC)
+                    .addModifier(FINAL)
+                    .addArgument("value",
+                            attributeMethodMapping.getKey().getType())
+                    .addCodeLine(String.format("this.container.putIfAbsent(\"%s\", value);",
+                            attributeMethodMapping.getKey().getName()))
+                    .addCodeLine("return this;"));
+        }
+    }
+
+    private void addSimpleSetters () {
+        for(Map.Entry<AttributeDescriptor, Optional<MethodDescriptor>> attributeMethodMapping:
+                this.annotatedClass.getAttributeSetterMapping().entrySet()) {
+            this.getMethods().add(new MethodDescriptor(
+                    computeSetterName(attributeMethodMapping.getKey().getName()), "void")
+                    .addModifier(PUBLIC)
+                    .addModifier(FINAL)
+                    .addArgument("value",
+                            attributeMethodMapping.getKey().getType())
+                    .addCodeLine(String.format("this.container.putIfAbsent(\"%s\", value);",
+                            attributeMethodMapping.getKey().getName())));
+        }
+    }
+
+    private void addBuildMethod () {
```

Comment:
this method is too long. Consider adding two util methods one to generate build method when there is no constructor or has no args,  and the other when there is constructor with params

Reply:
Done. Revised in [pull request 8](https://github.com/OscarCamacho/simpleannotation/pull/8)